### PR TITLE
Configuring Console Json Formatting with JsonWriterOptions doesn't work with Trimming

### LINKED ISSUE #73822: Configuring Console Json Formatting with JsonWriterOptions doesn't work with Trimming
### Description

We are suppressing a trimming warning here:

https://github.com/dotnet/runtime/blob/d8226260c9630c2fb196acaa6af7f5dbb45685ee/src/libraries/Microsoft.Extensions.Logging.Console/src/ConsoleLoggerExtensions.cs#L25-L27

But we aren't taking the correct steps to ensure all the properties that can be configured through the IConfiguration binder are being preserved. For example, properties on `JsonWriterOptions` can be trimmed causing the configuration to stop working.

### Reproduction Steps

Publish the following app both with and without trimming
`dotnet publish -r win-x64`
`dotnet publish -r win-x64 -p:PublishTrimmed=true`

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.Logging.Console" Version="6.0.0" />
  </ItemGroup>
</Project>
```

```C#
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.Logging;

IServiceCollection services = new ServiceCollection();

IConfiguration config = new ConfigurationBuilder()
    .AddInMemoryCollection(
        new[]
        {
            new KeyValuePair<string, string>("Logging:Console:FormatterName", "json"),
            new KeyValuePair<string, string>("Logging:Console:FormatterOptions:JsonWriterOptions:Indented", "true"),
        })
    .Build();

services.AddLogging(logging =>
{
    logging.AddConfiguration(config.GetSection("Logging"));
    logging.AddConsole();
});

var sp = services.BuildServiceProvider();

var logger = sp.GetRequiredService<ILogger<int>>();

logger.LogCritical("Press any key to continue");

Console.ReadKey();
```

### Expected behavior

The output of both apps is identical. Trimming shouldn't affect the behavior of the app.

### Actual behavior

When untrimmed, the output is:

```
{
  "EventId": 0,
  "LogLevel": "Critical",
  "Category": "int",
  "Message": "Press any key to continue",
  "State": {
    "Message": "Press any key to continue",
    "{OriginalFormat}": "Press any key to continue"
  }
}
```

When trimmed, the output is:

```
{"EventId":0,"LogLevel":"Critical","Category":"int","Message":"Press any key to continue","State":{"Message":"Press any key to continue","{OriginalFormat}":"Press any key to continue"}}
```

### Regression?

No

### Known Workarounds

You can manually preserve all the properties on `JsonWriterOptions`.

### Configuration

_No response_

### Other information

_No response_

### PR DESCRIPTION
**Pull request by eerhardt**
Ensure the correct members are preserved when trimming and add a trimming test.

Fix #73822 #73436

### CODE REVIEW

**Review by tarekgh — APPROVED** [review]
LGTM.

File: src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerExtensionsTests.cs [hunk 1]
```diff
@@ -271,7 +271,6 @@ public void AddSystemdConsole_OutsideConfig_TakesProperty()
         }
 
         [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
-        [ActiveIssue("https://github.com/dotnet/runtime/issues/73436", typeof(PlatformDetection), nameof(PlatformDetection.IsNativeAot))]
```
[the remark this answers is not in the document]
Comment (MichalStrehovsky) [reply src/libraries/Microsoft.Extensions.Logging.Console/tests/Microsoft.Extensions.Logging.Console.Tests/ConsoleLoggerExtensionsTests.cs:274]: Kicked off a test pass off. We only run a small subset of libs test by default and Microsoft.Extensions is not in the subset.

**Comment by eerhardt** [comment]
`runtime (Libraries Test Run release mono *` failures are: https://github.com/dotnet/runtime/issues/76019

NativeAOT failures are: https://github.com/dotnet/runtime/issues/76018

**Pull request merged**